src/Widgets/WindowCloneContainer.vala

```diff
@@ -254,8 +253,30 @@ namespace Gala {
 
             WindowClone? closest = null;
 
-            if (current_window == null) {
-                closest = (WindowClone) get_child_at_index (0);
+            // If current_window is not active it doesn't have the visible keyboard focus border
+            // so in order to not confuse the user we start from the beginning
+            if (!current_window.active || current_window == null) {
+                bool top_left = direction == UP || direction == LEFT;
+                int current_x = top_left ? int.MAX : 0;
+                int current_y = top_left ? int.MAX : 0;
+
+                foreach (unowned var child in get_children ()) {
+                    var window_rect = ((WindowClone) child).slot;
+
+                    if (top_left) {
+                        if (window_rect.x <= current_x && window_rect.y <= current_y) {
+                            current_x = window_rect.x;
+                            current_y = window_rect.y;
+                            closest = (WindowClone) child;
+                        }
+                    } else {
+                        if (window_rect.x >= current_x && window_rect.y >= current_y) {
+                            current_x = window_rect.x;
+                            current_y = window_rect.y;
+                            closest = (WindowClone) child;
+                        }
```

Comment:
This code actually selects the right bottom window instead of second from top/left.

Reply:
Yeah that was kinda my intention. Like when you have a menu and press down the top most is selected and when you press up the bottom most is selected. But I guess that's a @elementary/ux question. Also maybe it should be swapped so that right or down selects top left and up or left selects bottom right?